itou/www/dashboard/views.py

```diff
@@ -363,3 +364,28 @@ def edit_user_notifications(request, template_name="dashboard/edit_user_notifica
     }
 
     return render(request, template_name, context)
+
+
+@login_required
+def api_token(request, template_name="dashboard/api_token.html"):
+    if request.user.is_siae_staff:
+        current_siae_pk = request.session.get(global_constants.ITOU_SESSION_CURRENT_SIAE_KEY)
+        siae = get_object_or_404(Siae, pk=current_siae_pk)
+        membership = request.user.siaemembership_set.get(siae=siae)
+        if not membership.is_admin:
+            raise PermissionDenied
+
+    elif request.user.is_prescriber:
+        raise PermissionDenied
+
+    else:
+        raise PermissionDenied
+
+    if request.method == "POST":
+        token = Token.objects.create(user=request.user)
+    else:
+        token = Token.objects.filter(user=request.user).first()
```

Comment:
On attend un seul token par utilisateur.
```suggestion
        token = Token.objects.filter(user=request.user).get()
```

Reply:
Humpf. Peut-être une bonne occasion pour introduire :

```python
from django.shortcuts import _get_queryset


  def get_or_none(cls, *args, **kwargs):
      qs = _get_queryset(cls)
      try:
          obj = qs.get(*args, **kwargs)
      except qs.model.DoesNotExist:
          obj = None
      return obj

Reply:
L'idée c'est de crasher s'il y en a 2, c'est ça ?

Reply:
Tout à fait.